vortex-expr/src/transform/match_between.rs

```diff
@@ -1,121 +1,132 @@
+    }
 
-            // Extract pairs of comparison of the form (left left_op eq) and (eq right_op right)
-            let (eq, left, left_op, right, right_op) =
-                if GetItem::is(lhs.lhs()) && lhs.lhs().eq(rhs.lhs()) {
-                    (
-                        lhs.lhs().clone(),
-                        lhs.rhs().clone(),
-                        lhs.op().swap(),
-                        rhs.rhs().clone(),
-                        rhs.op(),
-                    )
-                } else if GetItem::is(lhs.lhs()) && lhs.lhs().eq(rhs.rhs()) {
-                    (
-                        lhs.lhs().clone(),
-                        lhs.rhs().clone(),
-                        lhs.op().swap(),
-                        rhs.lhs().clone(),
-                        rhs.op().swap(),
-                    )
-                } else if GetItem::is(lhs.rhs()) && lhs.rhs().eq(rhs.lhs()) {
-                    (
-                        lhs.rhs().clone(),
-                        lhs.lhs().clone(),
-                        lhs.op(),
-                        rhs.rhs().clone(),
-                        rhs.op(),
-                    )
-                } else if GetItem::is(lhs.rhs()) && lhs.rhs().eq(rhs.rhs()) {
-                    (
-                        lhs.rhs().clone(),
-                        lhs.lhs().clone(),
-                        lhs.op(),
-                        rhs.lhs().clone(),
-                        rhs.op().swap(),
-                    )
-                } else {
-                    return Ok(TransformResult::no(node));
-                };
-
-            // Find the greater op.
-            let (Some(left_lit), Some(right_lit)) =
-                (Literal::maybe_from(&left), Literal::maybe_from(&right))
-            else {
-                return Ok(TransformResult::no(node));
-            };
+    // Extract pairs of comparison of the form (left left_op eq) and (eq right_op right)
+    let (eq, left, left_op, right, right_op) = if GetItem::is(lhs.lhs()) && lhs.lhs().eq(rhs.lhs())
+    {
+        (
+            lhs.lhs().clone(),
+            lhs.rhs().clone(),
+            lhs.op().swap(),
+            rhs.rhs().clone(),
+            rhs.op(),
+        )
+    } else if GetItem::is(lhs.lhs()) && lhs.lhs().eq(rhs.rhs()) {
+        (
+            lhs.lhs().clone(),
+            lhs.rhs().clone(),
+            lhs.op().swap(),
+            rhs.lhs().clone(),
+            rhs.op().swap(),
+        )
+    } else if GetItem::is(lhs.rhs()) && lhs.rhs().eq(rhs.lhs()) {
+        (
+            lhs.rhs().clone(),
+            lhs.lhs().clone(),
+            lhs.op(),
+            rhs.rhs().clone(),
+            rhs.op(),
+        )
+    } else if GetItem::is(lhs.rhs()) && lhs.rhs().eq(rhs.rhs()) {
+        (
+            lhs.rhs().clone(),
+            lhs.lhs().clone(),
+            lhs.op(),
+            rhs.lhs().clone(),
+            rhs.op().swap(),
+        )
+    } else {
+        return None;
+    };
```

Comment:
this block is extremely hard to parse

Reply:
Any ideas how to clean up? Its a simple pattern match, but the lack of enums make it difficult?